lib/IRGen/GenType.cpp

```diff
@@ -423,6 +423,19 @@ static llvm::Value *computeExtraTagBytes(IRGenFunction &IGF, IRBuilder &Builder,
   return phi;
 }
 
+static Address alignAddress(IRGenFunction &IGF,
+		            Address originAddress,
+			    llvm::Value *shift){
+  auto &Builder = IGF.Builder;
+  auto &IGM = IGF.IGM;
+  auto *alignedAddress = Builder.CreateBitCast(originAddress, IGM.Int8PtrTy).getAddress();
+  if (IGM.Triple.getOS() == llvm::Triple::Linux && !IGM.Triple.isLittleEndian()) {
+    alignedAddress = Builder.CreateConstGEP1_32(alignedAddress, 4);
```

Comment:
`CreateConstByteArrayGEP` or similar here.  These typeless GEP constructors oughta be marked deprecated...

Reply:
> `CreateConstByteArrayGEP` or similar here.

Reply:
Do you mean to replace `CreateConstGEP1_32` with `CreateConstByteArrayGEP`? Any reasons?

Reply:
The untyped version of this function is an artifact of an API migration AFAICT, so at the very least you should call the overload passing an explicit type.  The other reason is that you can continue to work with `Address` throughout this helper function which has the added benefit of retaining the alignment of the original buffer.

Reply:
Thanks for the feedback. What about the following changes?
```swift
static Address alignAddress(IRGenFunction &IGF, Address originAddress,
                            llvm::Value *shift) {
  auto &Builder = IGF.Builder;
  auto &IGM = IGF.IGM;
  auto *alignedAddressValue =
      Builder.CreateBitCast(originAddress, IGM.Int8PtrTy).getAddress();
  Address alignedAddress = Builder.CreateBitCast(originAddress, IGM.Int8PtrTy);
  if (IGM.Triple.getOS() == llvm::Triple::Linux &&
      !IGM.Triple.isLittleEndian()) {
    alignedAddress = Builder.CreateConstByteArrayGEP(alignedAddress, Size(4));
    alignedAddressValue = alignedAddress.getAddress();
    alignedAddressValue =
        Builder.CreateGEP(alignedAddressValue, Builder.CreateNeg(shift));
    alignedAddress = Address(alignedAddressValue, alignedAddress.getAlignment());
  }
  return alignedAddress;
}
```